misk/src/main/kotlin/misk/web/WebTab.kt

```diff
@@ -10,22 +10,19 @@ abstract class WebTab(
   val capabilities: Set<String> = setOf(),
   val services: Set<String> = setOf()
 ) : ValidWebEntry(slug = slug, url_path_prefix = url_path_prefix) {
-  fun isAuthenticated(caller: MiskCaller?): Boolean {
-    return when {
-      capabilities.isEmpty() && services.isEmpty() -> true // no capabilities/service requirement => unauthenticated requests allowed (including when caller is null)
-      caller == null -> false // capability/service requirement present but caller null => assume authentication broken
-      capabilities.any { caller.capabilities.contains(it) } -> true // matching capability
-      services.any { caller.service == it } -> true // matching service
-      else -> false
-    }
+  fun isAuthenticated(caller: MiskCaller?): Boolean = when {
```

Comment:
any particular reason we have to re-implement our authentication logic for web tabs vs web actions? Would be good to have a single source of truth for what is pretty critical security logic

Reply:
Good point, we can reused the `isAllowed` function from `AccessInterceptor`. Will do so in a follow up PR.